src/components/rich-text-edit/adapter/vue-froala.ts

```diff
@@ -0,0 +1,479 @@
+// This code is largery borrowed from https://github.com/froala/vue-froala-wysiwyg.
+// However some changes have been made to "inputify" the froala editor and render is compatible with modUL input-style.
+import $ from 'jquery';
+import Vue from 'vue';
+import Component from 'vue-class-component';
+import { Prop, Watch } from 'vue-property-decorator';
+
+import { eraseTags, filterByTag, filterImg, filterNewLines, replaceTag, replaceTags } from '../../../utils/filter/htmlFilter';
+import WithRender from './vue-froala.html?style=./vue-froala.scss';
+
+require('froala-editor/js/froala_editor.pkgd.min');
+require('froala-editor/css/froala_editor.pkgd.min.css');
+require('froala-editor/js/languages/fr.js');
+
+const SPECIAL_TAGS: string[] = ['img', 'button', 'input', 'a'];
+
+const INNER_HTML_ATTR: string = 'innerHTML';
+
+enum froalaEvents {
+    Initialized = 'froalaEditor.initialized',
+    Focus = 'froalaEditor.focus',
+    Blur = 'froalaEditor.blur',
+    KeyUp = 'froalaEditor.keyup',
+    KeyDown = 'froalaEditor.keydown',
+    PasteAfter = 'froalaEditor.paste.after',
+    PasteBeforeCleanup = 'froalaEditor.paste.beforeCleanup',
+    WordPasteBefore = 'froalaEditor.paste.wordPaste.before',
+    CommandAfter = 'froalaEditor.commands.after'
+}
+
+enum froalaCommands {
+    FullScreen = 'fullscreen'
+}
+
+function cleanHtml(html: string): string {
+    // delete new lines so the regexps will work
+    html = filterNewLines(html);
+    // delete images
+    html = filterImg(html);
+    // delete videos
+    html = filterByTag('video', html);
+    // delete tables
+    html = eraseTags(['table', 'tbody', 'tr'], html);
+    html = replaceTag('td', 'p', html);
+    // replace headings with paragraphs
+    html = replaceTags(['h1', 'h2', 'h3', 'h4', 'h5', 'h6'], 'p', html);
+
+    return html;
+}
+
+class PopupPlugin {
+
+    private editor;
+    private buttonName;
+    private pluginName;
+    private icon;
+    private buttonList;
+
+    constructor(editor, name: string, icon: string, buttonList: string[]) {
+        this.editor = editor;
+        this.buttonName = `${name}Popup`;
+        this.pluginName = `${name}Plugin`;
+        this.icon = icon;
+        this.buttonList = buttonList;
+    }
+    initPopup(): void {
+        // Popup buttons.
+        let buttons: string = (this.buttonList.length > 1) ? `<div class="fr-buttons">${this.editor.button.buildList(this.buttonList)}</div>` : '';
+
+        // Load popup template.
+        let template: any = {
+            buttons: buttons,
+            custom_layer: ''
+        };
+
+        // Create popup.
+        return this.editor.popups.create(`${this.pluginName}.popup`, template);
+    }
+
+    // Show the popup
```

Comment:
@gabra20  je pense que tu as copy / paste du code?  un commentaire show the popup au dessus d'une fonction showPopup >.<

Reply:
Revérifier les autres commentaires autour de ce code aussi